crates/turborepo-ffi/src/cache.rs

```diff
@@ -0,0 +1,88 @@
+use turbopath::AbsoluteSystemPathBuf;
+use turborepo_api_client::APIClient;
+use turborepo_cache::{http::HttpCache, signature_authentication::ArtifactSignatureAuthenticator};
+
+use crate::{proto, Buffer};
+
+#[no_mangle]
+#[tokio::main]
+pub async extern "C" fn retrieve(buffer: Buffer) -> Buffer {
+    let req: proto::RetrieveRequest = match buffer.into_proto() {
+        Ok(req) => req,
+        Err(err) => {
+            let resp = proto::RetrieveResponse {
+                response: Some(proto::retrieve_response::Response::Error(err.to_string())),
+            };
+            return resp.into();
+        }
+    };
+
+    let http_cache_config = req.http_cache.expect("http_cache is required field");
+
+    let api_client_config = http_cache_config
+        .api_client
+        .expect("api_client is required field");
```

Comment:
protobuf makes non-primitive fields optional for some reason

Reply:
Fun fact: current best practice for protobuf is that all fields are intended to be optional. it's for forward and backward compatibility between different versions of services